rfc/rfc-61/rfc-61.md

```diff
@@ -0,0 +1,240 @@
+    * The space usage becomes (1 + (t-1) * p)/t
+    * Incremental use reduces the amount of data involved in the calculation
+
+When using snapshot view storage, for some scenarios where the proportion of changing data is not large, a better storage saving effect will be achieved
+We have a simple formula here to calculate the effect
+P indicates the proportion of changed data, and t indicates the number of time periods to be saved
+The lower the percentage of changing data, the better the storage savings
+So There is also a good savings for long periods of data
+
+At the same time, it has benefit for incremental computing resource saving
+
+3. Some typical scenarios
+   1. Every day generate a new snapshot base on original Hudi table which named tbl-YYYYMMDD, user can use snapshot table to generate derived tables,
+   provide report data. if user's downstream calculation logic changed, can choose relevant snapshot to re-process.
+   user also can set retain days as X day, clean out-of-date data automatically. SCD-2 should also can be achieved here.
+   2. One archived branch named yyyy-archived can be generated after compress and optimize. if our retention policy has been 
+   changed(let's say remove some sensitive information), then can generate a new snapshot base on this branch after operation done.
+   3. One Snapshot named pre-prod can release to customer after some quality validations passed base on any external tools.
+   
+## Implementation
+
+![basic_arch](basic_arch.png)
+
+### Extend Savepoint meta 
+Snapshot view need to extend the savepoint metadata, so we are going to add one struct with four fields: 
+* tag_name: tag name for your snapshot
+* retain-days: number of day, So data belongs to this snapshot will be retained for retain-days,  then can be clean after snapshot expire
+* database: database name in Catalog
+* table-name: table name in Catalog 
+
+new Savepoint Metadata should look like below:
+``` diff
+{
+	"type": "record",
+	"name": "HoodieSavepointMetadata",
+	"namespace": "org.apache.hudi.avro.model",
+	"fields": [{
+		"name": "savepointedBy",
+		"type": {
+			"type": "string",
+			"avro.java.string": "String"
+		}
+	}, {
+		"name": "savepointedAt",
+		"type": "long"
+	}, {
+		"name": "comments",
+		"type": {
+			"type": "string",
+			"avro.java.string": "String"
+		}
+	}, {
+		"name": "partitionMetadata",
+		"type": {
+			"type": "map",
+			"values": {
+				"type": "record",
+				"name": "HoodieSavepointPartitionMetadata",
+				"fields": [{
+					"name": "partitionPath",
+					"type": {
+						"type": "string",
+						"avro.java.string": "String"
+					}
+				}, {
+					"name": "savepointDataFile",
+					"type": {
+						"type": "array",
+						"items": {
+							"type": "string",
+							"avro.java.string": "String"
+						}
+					}
+				}]
+			},
+			"avro.java.string": "String"
+		}
+	}, {
+		"name": "version",
+		"type": ["int", "null"],
+		"default": 1
++	}, {
++		"name": "tag",
```

Comment:
i don't think it needs a nested structure here. its properties are fine to stay at top level